### DISCUSSION TITLE
'MutableRefObject<FlatList<any> | undefined>' is not assignable to type 'LegacyRef<FlatList<DataType>> | undefined'.

### DISCUSSION THREAD

**Discussion by NguyenHoangMinhkkkk** [software-mansion/react-native-reanimated#4725] [Q&A] [answered]
does someone know solution ?

//createRef not works for current.getNode().scrollToIndex but useRef work well with current.scrollToIndex

//const flatListRef = React.createRef<Animated.FlatList<TDataType>>();   
const flatListRef = React.useRef<FlatList>();

<Animated.FlatList ref={flatListRef} />

dependencies: 
"react-native": "0.71.8"
 "react-native-reanimated": "^3.2.0"

**Comment by tjzel** [comment]
Hi @NguyenHoangMinhkkkk. What is the exact version of `react-native-reanimated` that you use?

**Reply by NguyenHoangMinhkkkk** [reply]
hi. 
versions are:
"react-native": "0.71.8"
"react-native-reanimated": "^3.2.0"

"typescript": "4.8.4"

**Reply by NguyenHoangMinhkkkk** [reply]
i tried to upgrade react-native-reanimated 3.3.0/ and nothing changed.

**Reply by tjzel** [reply]
You actually should use `useAnimatedRef` from `react-native-reanimated` when trying to ref Animated components. Please tell me if using it fixes your problem.

**Comment by gradaom** [comment, marked as answer]
It seems that you are encountering an issue with using `createRef` for the `scrollToIndex` method in React Native, but `useRef` works well with it. 

In React Native, the `createRef` function creates a ref that can be used to access DOM elements or components. However, it may not work as expected with certain methods or properties of third-party libraries.

To resolve this issue, you can continue using `useRef` instead of `createRef`. Here's how you can modify your code:

```jsx
const flatListRef = React.useRef();

<Animated.FlatList ref={flatListRef} />
```

With this change, you can now use `flatListRef.current.scrollToIndex` without any issues.

Please make sure that you have the necessary versions of React Native and React Native Reanimated installed. Checking for compatibility among different packages and versions is important to ensure smooth functioning.

If you need further assistance or have any more questions, feel free to ask!

**Reply by NguyenHoangMinhkkkk** [reply]
![Screenshot 2023-07-18 at 14 12 17](https://github.com/software-mansion/react-native-reanimated/assets/30792140/736577a9-ef58-429b-a788-af1b4d740e2d)


it works but TS error, could you please help.

**Reply by tjzel** [reply]
@NguyenHoangMinhkkkk you should do

```ts
const ref = useAnimatedRef<Animated.FlatList>();
```
